identity/src/main/java/com/stripe/android/identity/ui/DocumentScanScreen.kt

```diff
@@ -184,51 +185,68 @@ private fun DocumentCaptureScreen(
     // Throttle the message change speed in Found state
     var foundResId by remember { mutableIntStateOf(R.string.stripe_hold_still) }
     var lastFoundResIdUpdateTime by remember { mutableLongStateOf(0L) }
+    var timeSinceInitalState by remember { mutableLongStateOf(0L) }
+    var shouldShowInitial by remember { mutableStateOf(false) }
     LaunchedEffect(documentScannerState) {
+        val currentTime = System.currentTimeMillis()
         if (documentScannerState is IdentityScanViewModel.State.Scanning &&
             documentScannerState.scanState is IdentityScanState.Found
         ) {
-            val currentTime = System.currentTimeMillis()
             if (currentTime - lastFoundResIdUpdateTime >= FEED_BACK_REFRESH_LIMIT) {
                 lastFoundResIdUpdateTime = currentTime
                 foundResId =
                     documentScannerState.scanState
                         .feedbackRes ?: R.string.stripe_hold_still
             }
         }
-    }
-
-    val message = when (documentScannerState) {
-        is IdentityScanViewModel.State.Scanning -> {
-            when (documentScannerState.scanState) {
-                is IdentityScanState.Finished -> stringResource(id = R.string.stripe_scanned)
-                is IdentityScanState.Found -> {
-                    stringResource(id = foundResId)
+        if (documentScannerState is IdentityScanViewModel.State.Scanning &&
+            documentScannerState.scanState is IdentityScanState.Initial
+        ) {
+            if (currentTime - timeSinceInitalState >= FEED_BACK_REFRESH_LIMIT) {
+                timeSinceInitalState = currentTime
+                launch {
+                    delay(FEED_BACK_REFRESH_LIMIT)
+                    shouldShowInitial = true
                 }
+            }
+        } else {
+            shouldShowInitial = false
+        }
+    }
 
-                is IdentityScanState.Initial -> {
-                    if (targetScanType.isNullOrFront()) {
-                        stringResource(id = R.string.stripe_position_id_front)
-                    } else {
-                        stringResource(id = R.string.stripe_position_id_back)
+    val message = if (shouldShowInitial) {
```

Comment:
It seems to me that a lot of this should live in a state in the ViewModel. Is this something you explored?

Reply:
pls see the reasoning above

Reply:
@tillh-stripe  as discussed off line, just moved the logic to IdentittyScanViewModel subclasses